### ISSUE TITLE
Show in local terminal should update PATH

### ISSUE THREAD

**Issue by rubensa** [angelozerr/angular2-eclipse#93]
If you:
  _Select project in Project Explorer ---> Right Click --> Show in local Terminal --> Terminal_
the new terminal windows does not have **PATH** variable updated.

When creating an **Angular Project**, there is issued a command that sets the **PATH** environment variable to include **node** path (I think there is **a bug here** as, at least if you have configured embedded **node.js**, seems to include **the path to node executable not to the bin directory** that contains **node** executable an **npm** link -so for creating the project is using my global configured node.js, not the specified "embedded" one-) and project **node_modules/.bin** directory.

I think that, if possible, this setup should be issued "always" when opening a terminal for the project or, at least, provide a way to do it "alone" (as you can "force" an **AngularCLI** run that also opens a new terminal with the **PATH** updated -in this case the path includes **the embedded node bin directory**, not only the executable, which I think is the right way-).

**Comment by rubensa** [comment]
From Terminal opened when creating the project (global node.js installed version):

~/development/eclipse/4.7.0_base/workspace/prueba$ node --version
**v6.11.0**


From Terminal opened when performing a "Run as" --> "ng build" (built in version):

~/development/eclipse/4.7.0_base/workspace/prueba$ node --version
**v6.9.4**

**Comment by angelozerr** [comment]
> (I think there is a bug here as, at least if you have configured embedded node.js, seems to include the path to node executable not to the bin directory

Tied to fix it, tell me if it's OK for you.

> If you:
Select project in Project Explorer ---> Right Click --> Show in local Terminal --> Terminal
the new terminal windows does not have PATH variable updated.

I know this problem. The problem is that TM terminal project doesn't provide an extension point kind to intialize PATH env before terminal is opened.

Perhaps we could create our own `Show In` to initialize PATH according the project settings?

**Comment by angelozerr** [comment]
> From Terminal opened when creating the project (global node.js installed version):

~/development/eclipse/4.7.0_base/workspace/prueba$ node --version
v6.11.0

From Terminal opened when performing a "Run as" --> "ng build" (built in version):

~/development/eclipse/4.7.0_base/workspace/prueba$ node --version
v6.9.4

Yes it's normal, launch initialize PATH with embed dir. You can check the PATH variable once terminal is opened.

**Comment by rubensa** [comment]
Just installed again from _http://oss.opensagres.fr/angular2-eclipse/1.3.0-SNAPSHOT/_ but same result...

On Terminal oppened to create the project:

~/development/eclipse/4.7.0_base/workspace/prueba$ node --version
**v6.11.0**
~/development/eclipse/4.7.0_base/workspace/prueba$ echo $PATH
/home/rubensa/development/eclipse/4.7.0_base/workspace/**.metadata/.plugins/ts.eclipse.ide.server.nodejs.embed.linux.gtk.x86_64/node-v6.9.4-linux-x64/bin/node**:/home/rubensa/development/eclipse/4.7.0_base/workspace/prueba/node_modules/.bin:/home/rubensa/software/node/bin:/home/rubensa/software/jdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

Could be network cache problem on my side or just not "really" fixed?

**Comment by angelozerr** [comment]
Update site was not built. Please wait for that following job https://opensagres.ci.cloudbees.com/job/angular2-eclipse/311/ is finished.

**Comment by rubensa** [comment]
Now it work like a charm!

Thanks @angelozerr !!

**NOTE**:  

> Today PATH are initialized by the wizard which opens the terminal. Perhaps we could create our own Show In to initialize PATH according the project settings?

I thinks It should be a good solution (as running a ng build is not so intuitive to open a new Terminal with the right PATH).  

**NOTE2**:  And what about "reuse" existing (if that is the case) Terminal used for previous execution and not opening a new one?

**Comment by angelozerr** [comment]
> I thinks It should be a good solution (as running a ng build is not so intuitive to open a new Terminal with the right PATH).

You don't like use launch to do `ng build`? Please not you can execute ng build by opening .angular-cli.json file

![image](https://user-images.githubusercontent.com/1932211/27962444-3a49bc46-6332-11e7-9ca3-71dc53593d65.png)

An another idea is to provide ng build, tests, etc as children of the .angular-cli.json  the project explorer like the scripts of package.json:

![image](https://user-images.githubusercontent.com/1932211/27962532-7cf3e5b2-6332-11e7-9967-32c75af55698.png)

Please create new issue if you like this idea.

>  And what about "reuse" existing (if that is the case) Terminal used for previous execution and not opening a new one?

It should work like this. If you redo a ng build, it should use the same terminal, no?

**Comment by rubensa** [comment]
> You don't like use launch to do ng build? 
No, I said that if I **only want to open a terminal** (but with the right PATHs set), It is not easy to guess that I must execute an ng command (using Eclipse).  Other way, If y open a new Terminal, the PATH is not set.

> It should work like this. If you redo a ng build, it should use the same terminal, no?
Not exactly.  I think that there should be one Terminal linked with one Angular Project.  If there is currently a linked Terminal opened, reuse.  If there is no one (never opened or just closed), create a new one and link with the project (for future re-use).  I'll create a new issue for this.

**Comment by Springrbua** [comment]
As there is now way to configure the Terminal, opened by "Show in Local Terminal", I guess it would be the best to provide a custom "Open in Terminal" for the plugins.  
This command could set the correct PATH and probably. It would also allow us to customize behavior in future, if needed, as we are in control of that terminal.

**Comment by rubensa** [comment]
New update on these.

It seems that some of the latter changes _broke_ the **PATH setup** as for now (when _Run As --> ng build_) It doesn't include current project "_node_modules/.bin_" directory in the **PATH**.  Only "_.metadata/.plugins/ts.eclipse.ide.server.nodejs.embed.linux.gtk.x86_64/node-v6.9.4-linux-x64/bin_" is included before current **PATH**.

Didn't notice the change till now that I closed all Terminal's and tried to open a new one.
